### PR TITLE
Dataset string memcap 3910/v2

### PR DESCRIPTION
**Pull request by inashivb**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/3910

Previous PR: https://github.com/OISF/suricata/pull/11047

Changes since v1:
- added length getter to THash API
- fixed string len for ip rep as well
- added doc note as this could be a breaking change for existing setups

### CODE REVIEW

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on build_asan.
  

Pipeline 20568

File: src/datasets.c [hunk 1]
```diff
@@ -1372,7 +1373,12 @@ static int DatasetAddString(Dataset *set, const uint8_t *data, const uint32_t da
     struct THashDataGetResult res = THashGetFromHash(set->hash, &lookup);
     if (res.data) {
         DatasetUnlockData(res.data);
-        return res.is_new ? 1 : 0;
+        if (res.is_new) {
+            THashIncrMemuse(set->hash, data_len);
```
Comment (victorjulien) [inline src/datasets.c:1377]: why is this needed? can we not use the getter inside the THashGetFromHash logic?

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on build_asan.
  

Pipeline 20572

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on build_asan.
  

Pipeline 20574

File: src/datasets.c (hunk 1 above)
Comment (inashivb) [reply src/datasets.c:1377]: Indeed we can and that's the better way. I did not try to make the existing code from the last PR better. My bad.
